snap/validate.go

```diff
@@ -476,6 +476,13 @@ func validateAppOrderNames(app *AppInfo, dependencies []string) error {
 	return nil
 }
 
+func validateAppWatchdog(app *AppInfo) error {
+	if app.WatchdogTimeout != 0 && !app.IsService() {
+		return fmt.Errorf("cannot define watchdog-timeout in application %q as it's not a service", app.Name)
+	}
```

Comment:
Should we also validate that WatchdogTimeout > 0 when IsService?

Reply:
Fixed, thanks for spotting this.